### PR TITLE
Add GitHub Action to check docs

### PR DESCRIPTION
**Pull request by Mogztter**
## Proposed Changes

1. Add GitHub Actions:

  - Build and check the documentation (i.e., make sure that the documentation does not contain broken links) on every pull requests
  - Notify Antora (`neo4j-documentation/docs-refresh`) when documentation changes are merged on the 4.0 branch (eventually the new/updated content will be published at http://neo4j.com)

2. Fix broken links

### CODE REVIEW

File: doc/docs/antora.yml [hunk 1]
```diff
@@ -10,7 +10,18 @@ asciidoc:
...
+    url-confluent-download: https://www.confluent.io/dowload/
+    url-confluent-kafka: https://docs.confluent.io/current/kafka
+    url-confluent-cloud: https://docs.confluent.io/current/cloud
+    url-confluent-install: https://docs.confluent.io/current/installation
+    url-confluent-quickstart: https://docs.confluent.io/platform/current/quickstart
+    url-confluent-clients: https://docs.confluent.io/platform/current/clients
+    url-confluent-ksql: https://docs.confluent.io/platform/current/ksqldb/index.html
+    url-confluent-java-client: https://docs.confluent.io/clients-kafka-java/current/overview.html
+    url-confluent-hub-neo4j: https://www.confluent.io/hub/neo4j/kafka-connect-neo4j
+    url-confluent-hub-datagen: https://www.confluent.io/hub/confluentinc/kafka-connect-datagen
```
Comment (Mogztter) [inline doc/docs/antora.yml:27]: I think it's easier to maintain if all the external URLs are defined in `antora.yml`.
Unfortunately, it's not possible to reference a value in YAML. In other words, we cannot do:

```yaml
url-confluent: https://www.confluent.io
url-confluent-blog: {url-confluent}/blog
```

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -4,6 +4,8 @@ on:
   push:
     branches:
       - '4.0'
+      # temporary: allow to see GH Actions result on fork
```
Comment (mroiter-larus) [inline .github/workflows/docs.yml:7]: @Mogztter Please, remember to remove this after tested

**Review by mroiter-larus — APPROVED** [review]

**Comment by mroiter-larus** [comment]
@moxious Looks good to me!

**Comment by conker84** [comment]
I'm merging this, thank you @Mogztter and @mroiter-larus !

File: .github/workflows/docs.yml (hunk 2 above)
Comment (Mogztter) [reply .github/workflows/docs.yml:7]: Ooops I didn't have time to remove it, I've opened: https://github.com/neo4j-contrib/neo4j-streams/pull/431

**Comment by Mogztter** [comment]
Yay! awesome :+1:

**Pull request merged**